src/engraving/dom/arpeggio.cpp

```diff
@@ -80,11 +80,17 @@ const TranslatableString& Arpeggio::arpeggioTypeName() const
 
 void Arpeggio::findAndAttachToChords()
 {
-    Chord* _chord = chord();
     track_idx_t strack = track();
+    Chord* _chord = chord();
+    Part* part = _chord->part();
+    track_idx_t btrack = part->endTrack();
+
+    if (track() + m_span > btrack) {
```

Comment:
Ideally, you should use strack here (instead of calling track()), and in other places as well